tests/unit/s2n_drbg_test.c

```diff
@@ -451,6 +451,8 @@ int main(int argc, char **argv)
     EXPECT_SUCCESS(s2n_stuffer_free(&nist_aes256_reference_entropy));
 
     EXPECT_SUCCESS(s2n_config_free(config));
+    /* Clean up with previously set functions */
+    POSIX_GUARD_RESULT(s2n_rand_cleanup());
```

Comment:
As mentioned in https://github.com/aws/s2n-tls/pull/4831#discussion_r1793967643, I'm not sure if this is the right fix for this. It seems better to properly initialize and cleanup the rand state with `s2n_init()` and `s2n_cleanup()` rather than rely on `s2n_rand_cleanup()` to reset the callbacks back to the defaults.

Reply:
Some callbacks require s2n_init called before they can be called. Therefore, I don't think this is the proper fix.